_tests/integration/fastlane_test.go

```diff
@@ -165,40 +161,48 @@ configs:
         workflow: primary
       workflows:
         deploy:
+          description: |
+            Deploys app using *Deploy to bitrise.io* step.
+
+            Next steps:
+            - Set up [Connecting to an Apple service with API key](https://devcenter.bitrise.io/en/accounts/connecting-to-services/connecting-to-an-apple-service-with-api-key.html##).
+            - Or further customise code signing following our [iOS code signing](https://devcenter.bitrise.io/en/code-signing/ios-code-signing.html) guide.
           steps:
           - activate-ssh-key@%s:
               run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
           - git-clone@%s: {}
           - cache-pull@%s: {}
-          - script@%s:
-              title: Do anything with Script step
-          - certificate-and-profile-installer@%s: {}
           - xcode-test@%s:
               inputs:
               - project_path: $BITRISE_PROJECT_PATH
               - scheme: $BITRISE_SCHEME
+              - test_repetition_mode: retry_on_failure
           - xcode-archive@%s:
               inputs:
               - project_path: $BITRISE_PROJECT_PATH
               - scheme: $BITRISE_SCHEME
               - distribution_method: $BITRISE_DISTRIBUTION_METHOD
-          - deploy-to-bitrise-io@%s: {}
+              - automatic_code_signing: api-key
           - cache-push@%s: {}
+          - deploy-to-bitrise-io@%s: {}
         primary:
+          description: |
+            The workflow executes the tests. And if there are none then only builds the project.
```

Comment:
NOTE: we know whether we found tests or not. Should not we provide a specific description?